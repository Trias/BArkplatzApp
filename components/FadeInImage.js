import React from 'react';
import { Animated } from 'react-native';

export default class FadeInImage extends React.Component {
    state = {
        opacity: new Animated.Value(0)
    };

    componentDidUpdate(prevProps) {
        if (prevProps.isFocused !== this.props.isFocused) {
            Animated.timing(this.state.opacity, {  
                toValue: 1,
                duration: 4000,
                useNativeDriver: true,
            }).start()
         
        }
      }

    componentDidMount() {
            Animated.timing(this.state.opacity, {  
            toValue: 1,
            duration: 4000,
            useNativeDriver: true,
        }).start()
    }

    render() {
        return (
            <Animated.Image
                onLoad={this.onLOad}
                {...this.props}
                style={[
                    {
                        opacity: this.state.opacity,
                        transform: [
                            {
                                scale: this.state.opacity.interpolate({
                                    inputRange: [0, 1],
                                    outputRange: [0.85, 1],
                                })
                            }
                        ]
                    },
                    this.props.style,
                ]}
            />
        )
    }
}