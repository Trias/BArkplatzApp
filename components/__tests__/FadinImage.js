import React from 'react';
import renderer from 'react-test-renderer';

import FadeInImage from '../FadeInImage';

jest.useFakeTimers()

it(`renders correctly`, () => {
  const tree = renderer.create(<FadeInImage 
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
