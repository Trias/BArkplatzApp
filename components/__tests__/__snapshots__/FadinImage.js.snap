// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  style={
    Object {
      "opacity": 0,
      "transform": Array [
        Object {
          "scale": 0.85,
        },
      ],
    }
  }
/>
`;
